### PR TITLE
New Post about APKBUILD.vim

### PR DESCRIPTION
**Pull request by maxice8**
Please review, this is very short @Cogitri

### CODE REVIEW

File: content/posts/08-meet-APKBUILD.vim.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: "APKBUILD.vim: filetype, plugins, syntax and plugins for APKBUILDs on vim"
+date: 2020-10-30T17:39:23-03:00
+draft: false
+---
+
+> Article by editor Leo <thinkabit.ukim@gmail.com>
+
+The Linux distro I contribute to, [Alpine Linux][1], uses build recipes called
+APKBUILDs, which are shell scripts but with context-aware variables that can
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
APKBUILDs. They are shell scripts but with context-aware variables that can
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+affect the final result of package and subject to policies that only apply to
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
affect the final result of package. They are subject to policies that only apply to
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+[Alpine Linux][1] and its packaging system `abuild`.
+
+[Vim][3] is a widely used text editor, many of our developers and users use it
+to edit APKBUILDs daily, but while vim has a filetype for shellscripts, it is
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
[Vim][3] is a widely used text editor, used by many of our developers and users
to edit APKBUILDs. But while vim has a filetype for shellscripts, it is
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+insufficient to provide syntax, plugins, indentation and other nice to have
+features that are specific to the context of an APKBUILD.
+
+In light of that I took upon myself to write a complete implementation of
+a vim `filetype`, specific for APKBUILDs.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
a vim `filetype` for APKBUILDs.
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+<!--more-->
+
+## The Whole Deal
+
+[APKBUILD.vim][2] is a plugin for Vim and [Neovim][4], that provides:
+
+1. Syntax Highlighting:  
+    1.1. Different colors for variables that affect `abuild` behavior.  
+    1.2. Error highlight for invalid values inside `abuild` variables, like non-existant arches in `arch`.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
    1.2. Error highlighting for invalid values inside `abuild` variables, like non-existant arches in `arch`.
```

Comment (Cogitri) [reply content/posts/08-meet-APKBUILD.vim.md]: Also: https://www.wordnik.com/words/nonexistent

File: content/posts/08-meet-APKBUILD.vim.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    1.4. Highlights `secfixes:` blocks, version identifiers and recognized security identifiers (CVE, XSA, etc).
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
    1.4. Highlighting of `secfixes:` blocks, version identifiers and recognized security identifiers (CVE, XSA, etc).  
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## The Whole Deal
+
+[APKBUILD.vim][2] is a plugin for Vim and [Neovim][4], that provides:
+
+1. Syntax Highlighting:  
+    1.1. Different colors for variables that affect `abuild` behavior.  
+    1.2. Error highlight for invalid values inside `abuild` variables, like non-existant arches in `arch`.
+    1.4. Highlights `secfixes:` blocks, version identifiers and recognized security identifiers (CVE, XSA, etc).  
+    1.5. Built on top of `sh.vim` so all syntax highlight from `filetype=sh` is there underneath.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
    1.5. It is built on top of `sh.vim`. That means all the syntax highlighting of `filetype=sh` is still in place.
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+2. Configuration:  
+    2.1. Sets correct `textwidth` of 80 columns.  
+    2.2: Sets `noexpandtab` as APKBUILD uses tabs for indentation.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
    2.2: Sets `noexpandtab` as APKBUILD use tabs for indentation.  
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+> have a different color than the ones normally used by shell.
+
+![Comparison of the main block](08-comparison-main-block.png)
+
+> Comparison of secfixes, notice how the `secfixes:` declaration, the versions and the security
+> identifiers are all highlighted.
+
+![Comparison of secfixes block](08-comparison-secfixes.png)
+
+> General comparison of Error highlighting.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
> General comparison of error highlighting.
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    1.2. Error highlight for invalid values inside `abuild` variables, like non-existant arches in `arch`.
+    1.4. Highlights `secfixes:` blocks, version identifiers and recognized security identifiers (CVE, XSA, etc).  
+    1.5. Built on top of `sh.vim` so all syntax highlight from `filetype=sh` is there underneath.
+2. Configuration:  
+    2.1. Sets correct `textwidth` of 80 columns.  
+    2.2: Sets `noexpandtab` as APKBUILD uses tabs for indentation.  
+    2.3: Sets `softtabstop` and `shiftwidth` to 4.
+3. [ALE][5] Integration:  
+    3.1. Provides a script to run [shellcheck][6] with APKBUILD aware filtering and register it into ALE.  
+    3.2. Add linters for `apkbuild-lint` and `secfixes-check` from [atools][7] and a handler for them.
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md]:
```suggestion
    3.2. Adds support for the linters `apkbuild-lint` and `secfixes-check` from [atools][7] and registers a handler for them.
```

**Review by Cogitri — DISMISSED** [review]

File: content/posts/08-meet-APKBUILD.vim.md (hunk 9 above)
Comment (maxice8) [reply content/posts/08-meet-APKBUILD.vim.md]: That is correct, Error is a syntax type in Vim

Comment (maxice8) [reply content/posts/08-meet-APKBUILD.vim.md]: `hi def link InvalidString Error` will make it bright red

Comment (Cogitri) [reply content/posts/08-meet-APKBUILD.vim.md]: Then it should probably be `Error`

Comment (maxice8) [reply content/posts/08-meet-APKBUILD.vim.md]: done

File: content/posts/08-meet-APKBUILD.vim.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
+---
+title: "APKBUILD.vim: filetype, plugins, syntax and plugins for APKBUILDs on vim"
+date: 2020-10-30T17:39:23-03:00
+draft: false
+---
+
+> Article by editor Leo <thinkabit.ukim@gmail.com>
+
+The Linux distro I contribute to, [Alpine Linux][1], uses build recipes called
+APKBUILDs, they are shell scripts but with context-aware variables that can
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md:10]:
```suggestion
APKBUILDs. They are shell scripts but with context-aware variables that can
```
Otherwise the sentence get really long and hard to follow

File: content/posts/08-meet-APKBUILD.vim.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+affect the final result of package and they aresubject to policies that only
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md:11]:
```suggestion
affect the final result of package. APKBUILDs are subject to policies that only
```

File: content/posts/08-meet-APKBUILD.vim.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+apply to [Alpine Linux][1] and its packaging system `abuild`.
+
+[Vim][3] is a text editor used by many of our developers and users use it
+to edit APKBUILDs. But while vim has a filetype for shellscripts, it is
```
Comment (Cogitri) [inline content/posts/08-meet-APKBUILD.vim.md:15]:
```suggestion
[Vim][3] is a text editor used by many of our developers and users for
editing APKBUILDs. But while vim has a filetype for shellscripts, it is
```
